lib/game_weapons.rb

```
@@ -0,0 +1,22 @@
+class GameWeapons
+
+  attr_reader :weapons
+
+  def initialize()
+    @weapons = [:rock, :paper, :scissors]
+  end
+
+
+  def random_weapon
+    weapons.sample
+  end
+
+  def select_weapon(weapon)
+    if weapons.include?(weapon)
+      return weapon
+    else
+      raise "Can't select that weapon"
```

As I said above I tend to avoid raising exceptions, as if this is not caught it will crash your whole app - I prefer to make sure in my tests that the wrong weapon won't be chosen, and if there is a danger I would use validations. 

But if you are going to raise here you should raise a custom exception so you can catch that specific error, the reason being otherwise if there were other genuine errors in this method your app would never catch them properly and you wouldn't realise you had a bug in your game